### PR TITLE
feat: Add newpipe support

### PR DESCRIPTION
**Pull request by SuhasDissa**
Added Newpipe support.
Only the search suggestions are tested to be working.

NewPipe is added as a replacement for PipedApi to make the testing easier

### CODE REVIEW

File: app/src/main/java/app/suhasdissa/mellowmusic/utils/ApiHelper.kt
Comment (Bnyro) [inline app/src/main/java/app/suhasdissa/mellowmusic/utils/ApiHelper.kt:1]: I guess that's from Piped? Can we convert that file to Kotlin?

Comment (SuhasDissa) [reply app/src/main/java/app/suhasdissa/mellowmusic/utils/ApiHelper.kt:1]: I renamed the `RetrofitHelper` to `ApiHelper`

Comment (Bnyro) [reply app/src/main/java/app/suhasdissa/mellowmusic/utils/ApiHelper.kt:1]: Sorry, I meant the DownloaderImpl

File: app/src/main/java/app/suhasdissa/mellowmusic/utils/DownloaderImpl.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                requestBuilder.header(headerName, headerValueList[0])
+            }
+        }
+        val response = client.newCall(requestBuilder.build()).execute()
+        if (response.code == 429) {
+            response.close()
+            throw ReCaptchaException("reCaptcha Challenge requested", url)
+        }
+        val body = response.body
+        val responseBodyToReturn: String? = body?.string()
```
Comment (Bnyro) [inline app/src/main/java/app/suhasdissa/mellowmusic/utils/DownloaderImpl.kt:47]:
```suggestion
        val body = response.body?.string()
```

**Comment by Bnyro** [comment]
Shouldn't we add a new tab or a preference for this to keep both working for now?

**Comment by SuhasDissa** [comment]
I'm not planning to add a preference switcher just yet.
I will consider adding a preference when this becomes more stable.

Feel free to add it yourself if you want.

**Review by Bnyro — APPROVED** [review]

**Comment by Bnyro** [comment]
> I'm not planning to add a preference switcher just yet. I will consider adding a preference when this becomes more stable.
> 
Makes sense, it's fine to add the preference for it just before the next release.

**Comment by SuhasDissa** [comment]
The last commit ([730a3e3](https://github.com/you-apps/MellowMusic/pull/23/commits/730a3e3ef52784f6604ff634298f02e830de84e1)) didn't work for me.
I don't see anything wrong with the code.

I Need some help with that

**Comment by Bnyro** [comment]
Fixed in https://github.com/you-apps/MellowMusic/pull/23/commits/ebcf9bd2a4f885367bc405869e9b6832e81d7d3e

**Pull request merged**